internal/transport/http2_client.go

```diff
@@ -1265,14 +1263,17 @@ func (t *http2Client) operateHeaders(frame *http2.MetaHeadersFrame) {
 	}
 
 	var (
-		// If a gRPC Response-Headers has already been received, then it means that the peer is speaking gRPC and we are in gRPC mode.
+		// If a gRPC Response-Headers has already been received, then it means
+		// that the peer is speaking gRPC and we are in gRPC mode.
 		isGRPC         = !initialHeader
 		mdata          = make(map[string][]string)
 		contentTypeErr string
 		grpcMessage    string
-		grpcStatus     *status.Status
 		statusGen      *status.Status
-		rawStatusCode  = codes.Unknown
+
+		grpcStatus string
```

Comment:
Maybe rename to `headerError` or something, with a comment like `// set if an error is encountered parsing headers`?

Reply:
Done